saemref/install.sls

```diff
@@ -63,6 +63,12 @@ legacy cleanup:
       - uwsgi
       - uwsgi-plugin-python
 
+{% if pillar.get('upgrade', False) %}
+drop old virtualenv:
+  file.absent:
+    - name: /home/{{ saemref.instance.user }}/venv
+{% endif %}
+
 venv:
   virtualenv.managed:
     - name: /home/{{ saemref.instance.user }}/venv
```

Comment:
I think it'd be safer to add a conditional `require: - file: drop old virtualenv` in this state.